curvefs/devops/filter_plugins/filter.py

```diff
@@ -69,22 +69,6 @@ def smart_topology(metaserver_groups, hostvars, nzone, ncopyset):
         'zonenum': nzone
     })
 
-    # metaservers
-    idx = 1
-    metaservers = []
```

Comment:
The `server` list can't generate when you delete the `metaserver` list, maybe you need delete the metaserver fileld of topology only.

Reply:
> The `server` list can't generate when you delete the `metaserver` list, maybe you need delete the metaserver fileld of topology only.

fixed